File: challenge-7/largest_element.c
/*
* C Program to print the largest element
* in an array
* Input Constraint: array_size >= 5
*/

#include <stdio.h>

// Returns the largest element from an integer array
int max(int *array, int size)
{
	// Assume that the first element is the largest
	int largest = array[0];

	for (int i = 1; i < size; i++)
	{
		// If current array element is greater than
		// currently largest element, current array element
		// becomes the new largest
		if (array[i] > largest)
		{
			largest = array[i];	
		}
	}

	return largest;
}

// Driver code
int main()
{
	int array_size;

	// Input until constraint satisfied
	do
	{
		printf("Input size: ");
		scanf("%d", &array_size);
	} while (array_size < 5);

	// Declaring variable sized array, available since C99
	int array[array_size];
	
	printf("Input array: \n");

	// Read array
	for (int i = 0; i < array_size; i++)
	{
		scanf("%d", &array[i]);
	}

	printf("The largest element is: %d\n", max(array, array_size));

}

File: challenge-7/LargestElement.java
/*
* Java program to print the largest element
* of an integer array
* Input Constraint: size_of_array >= 5
*/

import java.util.Scanner;

public class LargestElement {

	// Returns the largest element from the array
	static int find_largest(int[] array) {
		
		// Assume largest to be 0
		int largestElement = array[0];

		for (int arrayElement : array) {
			
			// If arrayElement is greater than current
			// value of largest element, replace.
			if (largestElement < arrayElement) {
				largestElement = arrayElement;
			}
		}

		return largestElement;
	}

	// Driver code
	public static void main(String[] args) {

		// Scanner object for input
		Scanner scanner = new Scanner(System.in);

		int arraySize;

		// Repeatedly take input until constraint satisfied
		do {
			System.out.print("Input size: ");
			arraySize = scanner.nextInt();
		} while (arraySize < 5);

		int[] array = new int[arraySize];

		System.out.println("Input array: ");

		// Read array
		for (int i = 0; i < array.length; i++) {
			array[i] = scanner.nextInt();
		}

		System.out.println("Largest element: " + find_largest(array));

	}
}